File: OOPS/16april2026/lecture1.js
class ATM {
    constructor(amount, name) {
        this.balance = amount;
        this.name = name;
    }

    deposit(amount) {
        this.balance += amount;
    }

    getData() {
        console.log(`Balance of ${this.name} is: ${this.balance}`);
    }
}

// Creating objects
let a = new ATM(100000, "yash");

// In JS, objects are reference-based → this is NOT a true copy
let b = Object.assign({}, a); // shallow copy

// But this loses class methods, so better way:
let bObj = new ATM(a.balance, a.name); // proper copy

bObj.deposit(1111);
bObj.getData();

File: OOPS/16april2026/lecture1.cpp
#include <iostream>
using namespace std;

class ATM {
private:
    double balance;
    string name;

public:
    // Constructor
    ATM(int amount, string name) {
        (*this).balance = amount;
        this->name = name;
    }

    void deposit(double amount) {
        balance += amount;
    }

    void getData() {
        cout << "Balance of "<<name<<" is: " << balance << endl;
    }
};

int main() {

    ATM a = ATM(100000,"yash"); 
    ATM b= ATM(a); // shallow copy 
    
    b.deposit(1111);
    b.getData();

    return 0;
}